### PR TITLE
Json

### PR DESCRIPTION
**Pull request by walde363**
Complete the week 2 step 3: JSON

### CODE REVIEW

File: portfolio/src/main/java/com/google/sps/servlets/HelloWorldServlet.java [hunk 1]
```diff
@@ -12,7 +12,17 @@
 
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    response.setContentType("text/html;");
-    response.getWriter().println("<h1>Hello Waldemar Ramos Colon</h1>");
+    String json = "{";
```
Comment (rjkroege) [inline portfolio/src/main/java/com/google/sps/servlets/HelloWorldServlet.java:15]: There is a much nicer way way to do this. Presumably you will do this in a subsequent PR / week when we discuss libraries?

File: portfolio/src/main/webapp/script.js [hunk 2]
```diff
@@ -30,8 +30,10 @@ function addRandomGreeting() {
 async function fetchCall() {
   const response = await fetch('/hello'); 
 
-  const responseText = await response.text(); 
+  const responsejson = await response.json(); 
+  
+  var index = Math.floor(Math.random() * 3);
```
Comment (rjkroege) [inline portfolio/src/main/webapp/script.js]: You should count the number of keys in `responsejson` and use that to scale the `index` to make sure that changes later in the Java code won't make line 38 have an error. 

Also: `index` can be const.

File: portfolio/src/main/webapp/script.js [hunk 3]
```diff
@@ -30,8 +30,10 @@ function addRandomGreeting() {
...
   const response = await fetch('/hello'); 
 
-  const responseText = await response.text(); 
+  const responsejson = await response.json(); 
+  
+  var index = Math.floor(Math.random() * 3);  
 
   const textContainer = document.getElementById('text-container'); 
-  textContainer.innerText = responseText; 
+  textContainer.innerHTML = responsejson[Object.keys(responsejson)[index]];
```
Comment (rjkroege) [inline portfolio/src/main/webapp/script.js:38]: This is clever but why did you change `innerText` to `innerHTML`?

**Review by rjkroege — CHANGES_REQUESTED** [review]
Looks great. I had some minor suggestions.

File: portfolio/src/main/webapp/script.js (hunk 3 above)
Comment (walde363) [reply portfolio/src/main/webapp/script.js:38]: I changed it because innerHTML returns the content in HTML format, the innerText returns a plain text.

File: portfolio/src/main/webapp/script.js (hunk 2 above)
Comment (walde363) [reply portfolio/src/main/webapp/script.js]: Done.

File: portfolio/src/main/java/com/google/sps/servlets/HelloWorldServlet.java (hunk 1 above)
Comment (walde363) [reply portfolio/src/main/java/com/google/sps/servlets/HelloWorldServlet.java:15]: Sorry, But I don't understand what you are trying to say.

Comment (rjkroege) [reply portfolio/src/main/java/com/google/sps/servlets/HelloWorldServlet.java:15]: A library like [gson](https://github.com/google/gson) would make it easier and less error prone to build JSON payloads. There is a week 3 lesson to use libraries. Converting this to `gson` then would be a nice improvement.

This comment requires no action on your part in this PR. It is a suggestion for future work.

File: portfolio/src/main/webapp/script.js (hunk 3 above)
Comment (rjkroege) [reply portfolio/src/main/webapp/script.js:38]: OK. I was being unclear. By assigning to `innerHTML`, you can have markup (and script) in the returned payload. But (to channel Spider Man) with great power comes great responsibility: the code that generates the string that you assign to an `innerHTML` block needs to always generate correct (and secure) HTML.

I don't think that you needed to make this change. There is no markup in the JSON payload and at some point, this could become a bug. It's also currently fine. In the future you should probably prefer to assign to `innerText` and use the DOM manipulation API.

**Review by rjkroege — APPROVED** [review]
